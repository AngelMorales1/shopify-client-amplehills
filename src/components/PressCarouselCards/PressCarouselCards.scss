@import 'settings';

.PressCarouselCards {
  &__card {
    border-radius: radius('s');
    margin: spacing(4);
    width: spacing(65);
    min-height: spacing(96);

    @media ($medium-up) {
      width: spacing(75);
    }
  }

  &__logo {
    width: spacing(21);
    height: spacing(10);
    object-fit: contain;
  }

  &__quote {
    font-size: $font-5;

    @media ($medium-up) {
      font-size: spacing(5.2);
    }
  }

  &__button {
    > span {
      margin-bottom: 1px;
    }
  }
}
