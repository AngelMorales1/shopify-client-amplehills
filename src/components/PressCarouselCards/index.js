import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import get from 'utils/get';
import cx from 'classnames';

import styles from './PressCarouselCards.scss';
import { Image, Button, HorizontalCarousel } from 'components/base';

const PressCarouselCards = ({ pressItemsInBlock, pressItems }) => {
  const fields = get(pressItems, '');
  const isCustomOrder = pressItemsInBlock.length;
  const selectedPressItems = isCustomOrder
    ? pressItemsInBlock
    : get(pressItems, 'fragments', []);
  let selectedPressItemsIds = selectedPressItems.map(fragment =>
    get(fragment[0], 'value', '')
  );

  return (
    <Fragment>
      {selectedPressItemsIds.map((pressItemId, i) => {
        const selectedPressItemsSimpleFragments = isCustomOrder
          ? pressItemsInBlock
          : get(pressItems, 'simpleFragments', {});
        const selectedPressItem = get(
          selectedPressItemsSimpleFragments,
          pressItemId,
          {}
        );

        return (
          <div
            key={pressItemId}
            className={cx(
              styles['PressCarouselCards__card'],
              'bg-white p3 flex flex-column justify-center items-center'
            )}
          >
            <Image
              className={cx(styles['PressCarouselCards__logo'])}
              src={get(selectedPressItem, 'logoImage.data', '')}
              alt={`${get(selectedPressItem, 'title', '')} logo`}
            />
            <span
              className={cx(
                styles['PressCarouselCards__quote'],
                'carter text-peach center py3'
              )}
            >
              {get(selectedPressItem, 'quote', '')}
            </span>
            <Button
              className={cx(
                styles['PressCarouselCards__button'],
                'uppercase detail'
              )}
              to={get(selectedPressItem, 'linkUrl', '')}
              label="Read about it"
              variant="primary-small"
              color="peach"
            />
          </div>
        );
      })}
    </Fragment>
  );
};

PressCarouselCards.propTypes = {
  block: PropTypes.shape({
    fields: PropTypes.shape({
      buttonLabel: PropTypes.string,
      buttonLink: PropTypes.string,
      pressItems: PropTypes.shape({
        fragments: PropTypes.arrayOf(PropTypes.array),
        simpleFragments: PropTypes.object
      }),
      customOrder: PropTypes.bool,
      drip: PropTypes.bool,
      sortByLatest: PropTypes.bool,
      title: PropTypes.string
    }),
    sys: PropTypes.shape({
      id: PropTypes.string
    })
  }),
  z: PropTypes.number,
  pressItems: PropTypes.shape({
    fragments: PropTypes.arrayOf(PropTypes.array),
    simpleFragments: PropTypes.object
  }),
  setRef: PropTypes.func
};

PressCarouselCards.defaultProps = {
  block: {
    fields: {
      buttonLabel: '',
      buttonLink: '',
      pressItems: {
        fragments: [],
        simpleFragments: {}
      },
      customOrder: false,
      drip: false,
      sortByLatest: true,
      title: ''
    },
    sys: {
      id: ''
    }
  },
  z: 0,
  pressItems: {
    fragments: [],
    simpleFragments: {}
  },
  setRef: () => {}
};

export default PressCarouselCards;
