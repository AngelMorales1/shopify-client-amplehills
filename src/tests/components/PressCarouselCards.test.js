import React from 'react';
import { shallow } from 'enzyme';

import PressCarouselCards from 'components/PressCarouselCards';

it('renders without data', () => {
  const component = shallow(<PressCarouselCards />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <PressCarouselCards
      pressItemsInBlock={{
        fragments: [
          [
            { key: 'foo', value: 'foo' },
            { key: 'foo', value: 'foo' },
            { key: 'foo', value: 'foo' },
            { key: 'foo', value: 'foo' },
            { key: 'foo', value: 'foo' }
          ]
        ],
        simpleFragments: { foo: {} }
      }}
      pressItems={{
        fragments: [
          [
            { key: 'foo', value: 'foo' },
            { key: 'foo', value: 'foo' },
            { key: 'foo', value: 'foo' },
            { key: 'foo', value: 'foo' },
            { key: 'foo', value: 'foo' }
          ]
        ],
        simpleFragments: { foo: {} }
      }}
    />
  );

  expect(component).toMatchSnapshot();
});
