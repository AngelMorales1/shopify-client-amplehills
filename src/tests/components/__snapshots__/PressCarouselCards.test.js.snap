// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PressCarouselCards
    pressItems={
        Object {
            "fragments": Array [
              Array [
                Object {
                  "key": "foo",
                  "value": "foo",
                },
                Object {
                  "key": "foo",
                  "value": "foo",
                },
                Object {
                  "key": "foo",
                  "value": "foo",
                },
                Object {
                  "key": "foo",
                  "value": "foo",
                },
                Object {
                  "key": "foo",
                  "value": "foo",
                },
              ],
            ],
            "simpleFragments": Object {
              "foo": Object {},
            },
          }
    }
    pressItemsInBlock={
        Object {
            "fragments": Array [
              Array [
                Object {
                  "key": "foo",
                  "value": "foo",
                },
                Object {
                  "key": "foo",
                  "value": "foo",
                },
                Object {
                  "key": "foo",
                  "value": "foo",
                },
                Object {
                  "key": "foo",
                  "value": "foo",
                },
                Object {
                  "key": "foo",
                  "value": "foo",
                },
              ],
            ],
            "simpleFragments": Object {
              "foo": Object {},
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="bg-white p3 flex flex-column justify-center items-center"
>
          <Image
                    alt=" logo"
                    bg={false}
                    className=""
                    onImgLoad={[Function]}
                    src=""
                    style={Object {}}
          />
          <span
                    className="carter text-peach center py3"
          />
          <Button
                    className="uppercase detail"
                    color="peach"
                    disabled={false}
                    fullWidth={false}
                    hover="shadow"
                    label="Read about it"
                    onClick={[Function]}
                    shadow={false}
                    to=""
                    type="button"
                    variant="primary-small"
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "foo",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Image
              alt=" logo"
              bg={false}
              className=""
              onImgLoad={[Function]}
              src=""
              style={Object {}}
/>,
            <span
              className="carter text-peach center py3"
/>,
            <Button
              className="uppercase detail"
              color="peach"
              disabled={false}
              fullWidth={false}
              hover="shadow"
              label="Read about it"
              onClick={[Function]}
              shadow={false}
              to=""
              type="button"
              variant="primary-small"
/>,
          ],
          "className": "bg-white p3 flex flex-column justify-center items-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": " logo",
              "bg": false,
              "children": null,
              "className": "",
              "onImgLoad": [Function],
              "src": "",
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "carter text-peach center py3",
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
              "className": "uppercase detail",
              "color": "peach",
              "disabled": false,
              "fullWidth": false,
              "hover": "shadow",
              "label": "Read about it",
              "onClick": [Function],
              "shadow": false,
              "to": "",
              "type": "button",
              "variant": "primary-small",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="bg-white p3 flex flex-column justify-center items-center"
>
            <Image
                        alt=" logo"
                        bg={false}
                        className=""
                        onImgLoad={[Function]}
                        src=""
                        style={Object {}}
            />
            <span
                        className="carter text-peach center py3"
            />
            <Button
                        className="uppercase detail"
                        color="peach"
                        disabled={false}
                        fullWidth={false}
                        hover="shadow"
                        label="Read about it"
                        onClick={[Function]}
                        shadow={false}
                        to=""
                        type="button"
                        variant="primary-small"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "foo",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Image
                alt=" logo"
                bg={false}
                className=""
                onImgLoad={[Function]}
                src=""
                style={Object {}}
/>,
              <span
                className="carter text-peach center py3"
/>,
              <Button
                className="uppercase detail"
                color="peach"
                disabled={false}
                fullWidth={false}
                hover="shadow"
                label="Read about it"
                onClick={[Function]}
                shadow={false}
                to=""
                type="button"
                variant="primary-small"
/>,
            ],
            "className": "bg-white p3 flex flex-column justify-center items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": " logo",
                "bg": false,
                "children": null,
                "className": "",
                "onImgLoad": [Function],
                "src": "",
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "carter text-peach center py3",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": null,
                "className": "uppercase detail",
                "color": "peach",
                "disabled": false,
                "fullWidth": false,
                "hover": "shadow",
                "label": "Read about it",
                "onClick": [Function],
                "shadow": false,
                "to": "",
                "type": "button",
                "variant": "primary-small",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PressCarouselCards
    pressItems={
        Object {
            "fragments": Array [],
            "simpleFragments": Object {},
          }
    }
    pressItemsInBlock={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
